viewing_party/party.py

```diff
@@ -1,23 +1,178 @@
-# ------------- WAVE 1 --------------------
-
+# 1st function in Wave 1
 def create_movie(title, genre, rating):
-    pass
 
-# -----------------------------------------
-# ------------- WAVE 2 --------------------
-# -----------------------------------------
+    if title is None or genre is None or rating is None:
+        return None
+
+    new_movie = {"title": title, "genre": genre, "rating": rating}
+
+    if isinstance(title, str) or isinstance(genre, str) or isinstance(rating, int):
+        return {
+            "title": title,
+            "genre": genre,
+            "rating": rating
+        }
+    else:
+        return None
+
+
+# 2nd function in Wave 1
+def add_to_watched(user_data, movie):
+
+    (user_data["watched"]).append(movie)
+    return user_data
+
+
+# 3rd function in Wave 1
+def add_to_watchlist(user_data, movie):
+
+    watchlist = user_data.get("watchlist", [])
+
+    watchlist.append(movie)
+    user_data["watchlist"] = watchlist
+    
+    return user_data
+
+
+# 4th function in Wave 1
+def watch_movie(user_data, title):
+
+    move_movie = False
+    index = 0
+    movie = None
+
+    for index in range(len(user_data["watchlist"])):
+        if title == user_data["watchlist"][index]["title"]:
+            move_movie = True
+            movie_index = index
+            movie = user_data["watchlist"][index]
+            print(user_data["watchlist"][index])
+    if move_movie:
+        (user_data["watched"]).append(movie)
+        del user_data["watchlist"][movie_index]
+    return user_data
+
+
+# 1st function in Wave 2
+def get_watched_avg_rating(user_data):
+
+    movie_count = 0
+    ratings_sum = 0
+    ratings_avg = 0
+
+    if user_data["watched"]:
+
+        for index in range(len(user_data["watched"])):
+
+            movie_count += 1
+            ratings_sum += user_data["watched"][index]["rating"]
+        ratings_avg = ratings_sum/movie_count
+    return ratings_avg
+
+
+# 2nd function in Wave 2
+def get_most_watched_genre(user_data):
+
+    most_watched_genre = user_data.get("watched",[])
+    
+    if not most_watched_genre: 
+        return None
+
+    genres = {}
+    for movie in most_watched_genre:
+        genre = movie.get("genre")
+        if genre in genres:
+            genres[genre] += 1
+        else:
+            genres[genre] = 1
+
+    most_watched_genre = max(genres, key=genres.get)
+
+    return most_watched_genre 
+
+
+# 1st function in Wave 3
+def get_unique_watched(user_data):
+
+    friends_watched_list = []
+    friends_watched_set = None
+    friends_list = []
+
+    for friend in user_data["friends"]:
+        for movie in friend["watched"]:
+            friends_watched_list.append(movie["title"])
+            friends_watched_set = set(friends_watched_list)
+
+    for movie in user_data["watched"]:
+        if movie["title"] not in friends_watched_set and movie not in friends_list:
+            friends_list.append(movie)
+
+    return friends_list

```

Comment:
In my eyes, `friends_list` isn't the most accurate name for this variable. I might propose `unique_movies_list`? Consider how this reads:

```python
def get_unique_watched(user_data):

    friends_watched_list = []
    friends_watched_set = None
    unique_movies_list = []

    # other stuff...

    for movie in user_data["watched"]:
        if movie["title"] not in friends_watched_set:
            if movie not in unique_movies_list:
                unique_movies_list.append(movie)

    return unique_movies_list
```